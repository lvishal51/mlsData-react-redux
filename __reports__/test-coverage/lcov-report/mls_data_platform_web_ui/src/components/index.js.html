<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
