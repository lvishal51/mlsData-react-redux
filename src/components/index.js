export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
