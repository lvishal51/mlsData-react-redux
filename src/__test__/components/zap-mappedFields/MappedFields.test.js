import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MappedFieldsComp as MappedFields }  from 'components';

const mappedFieldPayload = {
  "fields": [{
      id: '7OtNhsQo9EL7OdyVImlf',
      tableSystemName: 'ADD0',
      standardName: 'null',
      shortName: 'Street Address',
      longName: 'Street Address',
      lookupName: 'Street Address',
      relevance: 100,
      active: false,
      resource: "Property",
      resourceStandardName: "Residential"
    },
    {
      id: 'LfN3mCZe8QiN89tqVn8x',
      tableSystemName: 'OFFICE_14',
      standardName: 'null',
      shortName: 'Zip',
      longName: 'Zip',
      lookupName: 'null',
      relevance: 0,
      active: false,
      resource: "Property",
      resourceStandardName: "Residential"
  },
  {
    id: 'FYrN2AtQrhaM13cepV4p',
    tableSystemName: 'selling_member_address',
    standardName: 'null',
    shortName: 'SellMbrAddy',
    longName: 'null',
    lookupName: 'null',
    relevance: 0,
    active: true,
    resource: "Property",
    resourceStandardName: "Residential"
  }],
  "function": {
    id: 1,
    enumValue: 'SUM_ALL',
    description: 'Sum of all numbers coming from raw columns data',
    active: true,
    lastUpdateTs: 1539791700000,
    createTs: 1539791700000,
    resource: "Property",
    resourceStandardName: "Residential"
  }
};

const canonicalFunctionsData = [
  {
    id: 1,
    enumValue: 'SUM_ALL',
    description: 'Sum of all numbers coming from raw columns data',
    active: true,
    lastUpdateTs: 1539791700000,
    createTs: 1539791700000
  },
  {
    id: 2,
    enumValue: 'CONCAT_ALL_WITH_COMMA',
    description: 'Concatenation of values coming from raw columns data',
    active: true,
    lastUpdateTs: 1539791700000,
    createTs: 1539791700000
  },
  {
    id: 3,
    enumValue: 'CUSTOM',
    description: 'Custom transformation to be executed against specific columns coming from raw columns data',
    active: true,
    lastUpdateTs: 1539791700000,
    createTs: 1539791700000
  }
];

const inputProps = {
  selectedField: "ADDRESS",
  MlsMappedField: {
    isResponseSuccess: false,
    payload: mappedFieldPayload,
  },
  MlsUpdateMappedField: {
    payload: [],
    isResponseSuccess: false,
  },
  MlsCanonicalFunctionsData: {
    payload: canonicalFunctionsData,
    isResponseSuccess: false
  },
  MlsAutoCompleteFunctionsData: {
    payload: [],
    isResponseSuccess: false
  },
  postMappedData: () => {},
  updateMappedData: () => {},
  getAutoCompleteFunctions: () => {},
  getCanonicalFunctions: () => {},
  selectedCanonicalField: "ADDRESS",
  transformPreviewData: () => {},
  MlsCanonicalData: {},
  getSampleFileData: () => {},
  resetMapping: () => {},
};

describe('Zap Mapped Fields component', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MappedFields  {...inputProps} />);
  });
  it('should render correctly.', () => {
    wrapper.setProps({MlsMappedField: { isResponseSuccess: true, payload: mappedFieldPayload}});
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('getCheckedFieldsOnLoad function should be triggered correctly', () => {
    wrapper.setProps({MlsUpdateMappedField: { isResponseSuccess: true, payload: []}});
    wrapper.setState({ filteredData: inputProps.MlsMappedField.payload.fields});
    const spy = jest.spyOn(wrapper.instance(), "getCheckedFieldsOnLoad");
    wrapper.instance().forceUpdate();
    wrapper.instance().getCheckedFieldsOnLoad();
    expect(spy).toBeCalled();
  });

  it('onCheckedChange function should be triggered correctly', () => {
    wrapper.setProps({MlsCanonicalFunctionsData: { isResponseSuccess: true, payload: canonicalFunctionsData}});
    wrapper.setState({ filteredData: inputProps.MlsMappedField.payload.fields,
      isCheckedUnchecked: [true, false, true]});
    const event = { target: { value: 'ADD0' }};
    const spy = jest.spyOn(wrapper.instance(), "onCheckedChange");
    wrapper.instance().forceUpdate();
    wrapper.find("[type='checkbox']").at(0).simulate('change', event);
    expect(spy).toBeCalled();

    const instance = wrapper.instance();
    instance.onCheckedChange(0, event); // for inactive field
    instance.onCheckedChange(2, event); // for active field

  });

  it('onSearchIconClick function should be triggered correctly', () => {
    wrapper.setProps({MlsAutoCompleteFunctionsData: { isResponseSuccess: true, payload: []}});
    const spy = jest.spyOn(wrapper.instance(), "onSearchIconClick");
    wrapper.instance().forceUpdate();
    wrapper.find("#field").simulate('click');
    expect(spy).toBeCalled();
  });

  it('modalToggle function should be triggered correctly', () => {
    const spy = jest.spyOn(wrapper.instance(), "modalToggle");
    wrapper.setState({
      availableFunctions: canonicalFunctionsData,
      receivedFunction: mappedFieldPayload.function,
      isCheckedUnchecked: [{
        id: '7OtNhsQo9EL7OdyVImlf',
        tableSystemName: 'ADD0',
        standardName: 'null',
        shortName: 'Street Address',
        longName: 'Street Address',
        lookupName: 'Street Address',
        relevance: 100,
        active: false
      },
      {
        id: 'LfN3mCZe8QiN89tqVn8x',
        tableSystemName: 'OFFICE_14',
        standardName: 'null',
        shortName: 'Zip',
        longName: 'Zip',
        lookupName: 'null',
        relevance: 0,
        active: false
      }
    ],
      expression: "SUM_ALL(ADD0)",
      selectedFunction: "DEFAULT",
    });
    wrapper.instance().forceUpdate();
    wrapper.find(".saveButton").simulate('click');
    expect(spy).toBeCalled();
  });

  it('toggleTooltip function should be called correctly', () => {
    wrapper.find('Tooltip').at(0).prop('toggle')();
    expect(wrapper.state('tooltipOpen')).toBe(true);
  });

  it('save button click', () => {
    wrapper.setState({ checkedFields: ['member_1','member_16'], selectedFunction: 'SUM_ALL', modal: true });
    wrapper.find('.saveButton').simulate('click');
  });

  it('test for getMappedFieldDescription function with not null data', () => {
    const spy = jest.spyOn(wrapper.instance(), "getMappedFieldDescription");
    wrapper.setState({ mappedField: mappedFieldPayload.fields[0] });
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('test for getMappedFieldDescription function with null longName', () => {
    const spy = jest.spyOn(wrapper.instance(), "getMappedFieldDescription");
    wrapper.setState({ mappedField: mappedFieldPayload.fields[2] });
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('test for getFilteredData function with string data', () => {
    const instance = wrapper.instance();
    instance.getFilteredData('SUM', 'SU');
  });

  it('test for getFilteredData function with object data', () => {
    const instance = wrapper.instance();
    instance.getFilteredData({ name: 'ADD0' }, 'AD');
  });

  it('test for filterExpressions function with mappedFields true', () => {
    const instance = wrapper.instance();
    instance.filterExpressions('sum', ['ADD0', 'Address', 'AddressInternet'], true, 'su');
  });

  it('test for filterExpressions function with mappedFields false and no subexpression', () => {
    const instance = wrapper.instance();
    instance.filterExpressions('sum', ['ADD0', 'Address', 'AddressInternet'], false, '');
  });

  it('getSampleFileData function should be triggered correctly', () => {
    const spy = jest.spyOn(wrapper.instance(), "getSampleFileData");
    wrapper.instance().forceUpdate();
    wrapper.find(".tableSystemName").at(0).simulate('click');
    expect(spy).toBeCalled();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance.handleInputChange('resource', 'Property', [{
      selectedKey: 'resource',
      selectedValue: 'Property'
    }]);
  });

  it('handleInputChange function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance.handleInputChange('class', 'A', [{
      selectedKey: 'class',
      selectedValue: 'A'
    }]);
  });

  it('handleInputChange function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance.handleInputChange('', '', [{
      selectedKey: 'k',
      selectedValue: '',
      selectedText: 're'
    }]);
  });

  it('handleInputChange function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance.handleInputChange('', '', [{
      selectedKey: '',
      selectedValue: '',
      selectedText: ''
    }]);
  });

  it('onSearchDesc function should be triggered correctly for given searchQuery', () => {
    const instance = wrapper.instance();
    const event = {
      target: {
        value: 'fea'
      }
    }
    instance.onSearchDesc(event);
  });

  it('onSearchDesc function should be triggered correctly for empty string', () => {
    const instance = wrapper.instance();
    const event = {
      target: {
        value: ''
      }
    }
    instance.onSearchDesc(event);
  });

  it('getSuggestionHeaderTitle function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance.getSuggestionHeaderTitle("className");
  });

  it('_renderSuggestionHeader function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance._renderSuggestionHeader(mappedFieldPayload.fields[0]);
  });

  it('_renderSuggestionItem function should be triggered correctly', () => {
    wrapper.setState({ autoSuggestKey: "resource"});
    const instance = wrapper.instance();
    instance._renderSuggestionItem(mappedFieldPayload.fields[0]);
  });

  it('updateQuery function should be triggered correctly', () => {
    const instance = wrapper.instance();
    instance._updateExpression("Test");
  });

  it('componentWillUnmount should be called on unmount', () => {
    const component = shallow(<MappedFields {...inputProps} />);
    const componentWillUnmount = jest.spyOn(component.instance(), 'componentWillUnmount');
    component.unmount();
    expect(componentWillUnmount).toHaveBeenCalled();
  });

});
