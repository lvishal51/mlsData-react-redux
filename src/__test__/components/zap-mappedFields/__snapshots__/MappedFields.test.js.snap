// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap Mapped Fields component should render correctly. 1`] = `
<div
  className="mappedFields"
>
  <Card
    onScroll={[Function]}
    tag="div"
  >
    <CardHeader
      className="cardHeaderWhite"
      tag="div"
    >
      <span
        className="h5 d-inline-block"
      >
        Mapped Fields
      </span>
    </CardHeader>
    <CardBody
      className="p-0"
      tag="div"
    >
      <div
        className="mappedFields"
      >
        <Row
          className="m-0 mappedHeaderRow"
          tag="div"
        >
          <Col
            className="mappedTableHeader"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="2"
          >
            Select
          </Col>
          <Col
            className="mappedTableHeader"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="3"
          >
            Field
            <div
              className="tableSearch"
            >
              <img
                className="tableSearchIcon"
                id="field"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
          </Col>
          <Col
            className="mappedTableHeader"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="5"
          >
            <span>
              DescriptionÂ 
              <i
                className="fa fa-info-circle"
                id="descriptionTooltip"
              />
            </span>
            <div
              className="tableSearch"
            >
              <img
                className="tableSearchIcon"
                id="desc"
                onClick={[Function]}
                src="test-file-stub"
              />
            </div>
            <Tooltip
              autohide={true}
              delay={
                Object {
                  "hide": 250,
                  "show": 0,
                }
              }
              hideArrow={false}
              isOpen={false}
              placement="top"
              placementPrefix="bs-tooltip"
              target="descriptionTooltip"
              toggle={[Function]}
            >
              Description is a combination of tableSystemName, longName and lookupName
            </Tooltip>
          </Col>
          <Col
            className="mappedTableHeader"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="2"
          >
            Relevancy in %
          </Col>
        </Row>
        <div
          className="mappedTable"
        >
          <Row
            className="m-0 mappedTableCell"
            key="FYrN2AtQrhaM13cepV4p"
            tag="div"
          >
            <Col
              className=""
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="2"
            >
              <FormGroup
                check={true}
                tag="div"
              >
                <Input
                  checked={true}
                  className="checkbox"
                  data-index={0}
                  id="checkbox_0"
                  onChange={[Function]}
                  type="checkbox"
                  value="selling_member_address"
                />
              </FormGroup>
            </Col>
            <Col
              className="tableSystemName"
              onClick={[Function]}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="3"
            >
              <span
                id="field_FYrN2AtQrhaM13cepV4p"
              >
                selling_member_address
              </span>
              <Tooltip
                autohide={true}
                className="customTooltip "
                delay={
                  Object {
                    "hide": 250,
                    "show": 0,
                  }
                }
                hideArrow={false}
                isOpen={false}
                placement="right"
                placementPrefix="bs-tooltip"
                target="field_FYrN2AtQrhaM13cepV4p"
                toggle={[Function]}
              >
                <span>
                  No Sample Data Available...
                </span>
              </Tooltip>
            </Col>
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="5"
            >
              selling_member_address 
            </Col>
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="2"
            >
              0
            </Col>
          </Row>
          <Row
            className="m-0 mappedTableCell"
            key="7OtNhsQo9EL7OdyVImlf"
            tag="div"
          >
            <Col
              className=""
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="2"
            >
              <FormGroup
                check={true}
                tag="div"
              >
                <Input
                  checked={false}
                  className="checkbox"
                  data-index={1}
                  id="checkbox_1"
                  onChange={[Function]}
                  type="checkbox"
                  value="ADD0"
                />
              </FormGroup>
            </Col>
            <Col
              className="tableSystemName"
              onClick={[Function]}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="3"
            >
              <span
                id="field_7OtNhsQo9EL7OdyVImlf"
              >
                ADD0
              </span>
              <Tooltip
                autohide={true}
                className="customTooltip "
                delay={
                  Object {
                    "hide": 250,
                    "show": 0,
                  }
                }
                hideArrow={false}
                isOpen={false}
                placement="right"
                placementPrefix="bs-tooltip"
                target="field_7OtNhsQo9EL7OdyVImlf"
                toggle={[Function]}
              >
                <span>
                  No Sample Data Available...
                </span>
              </Tooltip>
            </Col>
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="5"
            >
              ADD0 |  Street Address
            </Col>
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="2"
            >
              100.00
            </Col>
          </Row>
          <Row
            className="m-0 mappedTableCell"
            key="LfN3mCZe8QiN89tqVn8x"
            tag="div"
          >
            <Col
              className=""
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="2"
            >
              <FormGroup
                check={true}
                tag="div"
              >
                <Input
                  checked={false}
                  className="checkbox"
                  data-index={2}
                  id="checkbox_2"
                  onChange={[Function]}
                  type="checkbox"
                  value="OFFICE_14"
                />
              </FormGroup>
            </Col>
            <Col
              className="tableSystemName"
              onClick={[Function]}
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="3"
            >
              <span
                id="field_LfN3mCZe8QiN89tqVn8x"
              >
                OFFICE_14
              </span>
              <Tooltip
                autohide={true}
                className="customTooltip "
                delay={
                  Object {
                    "hide": 250,
                    "show": 0,
                  }
                }
                hideArrow={false}
                isOpen={false}
                placement="right"
                placementPrefix="bs-tooltip"
                target="field_LfN3mCZe8QiN89tqVn8x"
                toggle={[Function]}
              >
                <span>
                  No Sample Data Available...
                </span>
              </Tooltip>
            </Col>
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="5"
            >
              OFFICE_14 |  Zip
            </Col>
            <Col
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
              xs="2"
            >
              0
            </Col>
          </Row>
        </div>
      </div>
    </CardBody>
  </Card>
  <section
    className="customSection"
  >
    <Row
      tag="div"
    >
      <Col
        lg="6"
        md="6"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <FormGroup
          tag="div"
        >
          <Label
            for="autocomplete"
            tag="label"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            Function Expression
          </Label>
          <Connect(AutoComplete)
            onChange={[Function]}
            selectedFields={
              Array [
                Object {
                  "active": true,
                  "id": "FYrN2AtQrhaM13cepV4p",
                  "longName": "null",
                  "lookupName": "null",
                  "relevance": 0,
                  "resource": "Property",
                  "resourceStandardName": "Residential",
                  "shortName": "SellMbrAddy",
                  "standardName": "null",
                  "tableSystemName": "selling_member_address",
                },
              ]
            }
            udfs={Array []}
          />
        </FormGroup>
      </Col>
      <Col
        className="d-flex"
        lg="3"
        md="3"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <Button
          className="previewButton zapButton align-self-center"
          color="secondary"
          disabled={true}
          onClick={[Function]}
          tag="button"
        >
          Preview
        </Button>
      </Col>
    </Row>
  </section>
  <section
    className="customSection"
  >
    <Card
      tag="div"
    >
      <CardHeader
        className="functionDetails"
        tag="div"
      >
        Function Details
      </CardHeader>
      <CardBody
        tag="div"
      >
        <div>
          <p>
            <span
              className="font-weight-bold"
            >
              Selected Fields: 
            </span>
            <span>
              selling_member_address
            </span>
          </p>
        </div>
      </CardBody>
    </Card>
    <div
      className="buttonWrapper"
    >
      <Button
        className="zapButton"
        color="secondary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      >
        Reset Mapping
      </Button>
      <Button
        className="saveButton zapButton"
        color="secondary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      >
        Save
      </Button>
    </div>
  </section>
</div>
`;
